be2-scala/src/test/scala/ch/epfl/pop/pubsub/graph/validators/SchemaVerifierSuite.scala

```diff
@@ -22,6 +22,21 @@ class SchemaVerifierSuite extends FunSuite with Matchers {
 
   import SchemaVerifierSuite._
 
+  /* ----------------------------- Test invalid JSON requests ----------------------------- */
+  test("Invalid jsonString - string instead of JSON object") {
+    val stringTest = "hey"
+
+    verifyRpcSchema(stringTest) shouldBe a[Right[_, PipelineError]]
+  }
+
+  test("Invalid jsonString - invalid object instead of JSON object") {
```

Comment:
I think that test is out-of-scope: forcing the Java/Scala runtime type of the json string to be improper would cause a `ClassCastException` even before this PR.